prow/cmd/deck/pr_history.go

```diff
@@ -201,11 +202,31 @@ func parsePullURL(u *url.URL) (org, repo string, pr int, err error) {
 }
 
 // getGCSDirsForPR returns a map from bucket names -> set of "directories" containing presubmit data
-func getGCSDirsForPR(config *config.Config, org, repo string, pr int) (map[string]sets.String, error) {
+func getGCSDirsForPR(config *config.Config, org, repo string, pr int, gc *git.Client, ghc *github.Client) (map[string]sets.String, error) {
```

Comment:
This seems backwards and some of that is from the original impl -- we should just read the PR-specific directory in GCS for this data instead of looking to see what presubmits are registered. Especially now that someone could have multiple iterations of the presubmits to run on their PR over the lifetime of the PR, but this is true today, too.

Reply:
https://github.com/kubernetes/test-infra/pull/10169#discussion_r238773107

Reply:
O would like to just keep the behaviour as-is for now and we can discuss separately if we should change the whole thing to fetch the data about which job existed for a PR from GCS @stevekuznetsov ?